src/Symfony/Component/Messenger/composer.json

```diff
@@ -17,12 +17,13 @@
     ],
     "require": {
         "php": "^7.2.5",
-        "psr/log": "~1.0"
+        "psr/log": "~1.0",
+        "symfony/messenger-amqp": "^5.1",
+        "symfony/messenger-doctrine": "^5.1",
+        "symfony/messenger-redis": "^5.1"
```

Comment:
these requires kinda defeat the need for separate packages IMHO.

We should not add them IMHO, as we can never remove them because of BC.

Alternatively keep the legacy transport in place, and let the new bridges extend from those meanwhile.

Reply:
We can (and will) remove them in 6.0.

Reply:
These transports should have been in Bridge namespace in the first place. I moving these now, since we are about to add more transports.

Reply:
so i can require symfony/messenger, while using this new symfony/messenger-amq transport, then updating to 6.0 my transport is removed .. and it crashes?

Reply:
see https://github.com/symfony/symfony/issues/29229 which triggered me. Maybe we can and should do it in this case.

Reply:
You are correct. 

Assuming you ignored deprecation messages, the changelog and the UPGRADE docs.

(I just added upgrade docs)

Reply:
im considering new users, they require symfony/messenger tomorrow, start using the new transports wihout any hint they actually should require it separately.

if we can remove those requires in 6.0 then it's fine, but im worried in Nov 2021 we'll consider it a BC break. In this case, not adding them now would avoid that.

Reply:
If you require `symfony/messenger` tomorrow and start using the transports, you will get a deprecation warning. The deprecation warnings are actually really nice:

> The "Symfony\Component\Messenger\Transport\Doctrine\DoctrineTransportFactory" class is deprecated since Symfony 5.1, use "Symfony\Component\Messenger\Bridge\Doctrine\Transport\DoctrineTransportFactory" instead. The Doctrine transport has been moved to package "symfony/messenger-doctrine" and will not be included by default in 6.0. Run "composer require symfony/messenger-doctrine".

So the deprecation layer is there - and (unless you ignore the deprecation warnings) you'll know to require the individual packages.

Reply:
are you sure @weaverryan? It triggers the deprecation **only** if you load the legacy class. Which is not the case for a simple usage when you let the Bundle loading the transport (see changes in src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.xml).

For a migration path shouldn't we:
- let code in `symfony/messenger`
- create a class in `symfony/bridge-transport` that reuse the code (trait or extends) (not an issue as the bridge have a already requires `symfony/messenger`)
- update the bundle to use `symfony/bridge-transport` class when exist, and fallback to `symfony/messenger`
- trigger a deprecation when people use class from `symfony/messenger`